core/src/main/scala/weaponregex/parser/Parser.scala

```diff
@@ -12,26 +12,41 @@ import scala.util.{Failure, Success, Try}
   */
 object Parser {
 
+  /** Apply the parser to parse the given pattern and flags
+    * @param pattern
+    *   The regex pattern to be parsed
+    * @param flags
+    *   The regex flags to be used
+    * @return
+    *   A `Success` of parsed [[weaponregex.model.regextree.RegexTree]] if can be parsed, a `Failure` otherwise
+    */
+  def apply(pattern: String, flags: String, flavor: ParserFlavor): Try[RegexTree] =
+    flavor match {
+      case ParserFlavorJVM => new ParserJVM(pattern).parse
+      case ParserFlavorJS  => new ParserJS(pattern, flags).parse
+      case _               => Failure(new RuntimeException("[Error] Parser: Unsupported regex flavor"))
+    }
+
   /** Apply the parser to parse the given pattern
     * @param pattern
     *   The regex pattern to be parsed
     * @return
     *   A `Success` of parsed [[weaponregex.model.regextree.RegexTree]] if can be parsed, a `Failure` otherwise
     */
-  def apply(pattern: String, flavor: ParserFlavor = ParserFlavorJVM): Try[RegexTree] = flavor match {
-    case ParserFlavorJVM => new ParserJVM(pattern).parse
-    case ParserFlavorJS  => new ParserJS(pattern).parse
-    case _               => Failure(new RuntimeException("[Error] Parser: Unsupported regex flavor"))
-  }
+  def apply(pattern: String, flavor: ParserFlavor = ParserFlavorJVM): Try[RegexTree] = apply(pattern, "", flavor)
 }
 
 /** The based abstract parser
   * @param pattern
   *   The regex pattern to be parsed
+  * @param flags
+  *   The regex flags to be used
   * @note
   *   The parsing rules methods inside this class is created based on the defined grammar
   */
-abstract class Parser(val pattern: String) {
+abstract class Parser(val pattern: String, val flags: String) {
```

Comment:
Since flags is only something that is relevant to the ParserJS, can the parameter be moved to only be added there? ParserJVM will never have a flags as it doesn't exist

Reply:
The `Pattern` in Java somehow seems to have support for “global” flags ([reference](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#compile(java.lang.String,int))). I also checked and many online regex testers for Java also have global flags support.

Reply:
Are they relevant to mutating? Especially since they're not used and have a very different implementation compared to js, I think it's cleaner to keep it in ParserJS. If we ever need to add flags support for jvm we can add them to ParserJVM

Reply:
Well, mutating is affected by the parsed result and global flags do affect parsing. For example, `abc #comment` without any flags matches the whole string `abc #comment`, but with the `COMMENT` global flag, it only matches `abc`, so their parse trees are different. But from what I can tell these flags only affect the parsing in a very small number of edge cases.

However, I agree that their implementations are different, so it make sense to move the current flags support to ParserJS